BraveWallet/Crypto/BuySendSwap/BuySendSwapDestination.swift

```diff
@@ -6,15 +6,26 @@
 import Foundation
 import struct Shared.Strings
 import SwiftUI
+import BraveCore
 
 /// Used to determine where a user is navigated to when they tap on a buy, send or swap button
-enum BuySendSwapDestination: String, Identifiable, CaseIterable {
-  case buy
-  case send
-  case swap
+enum BuySendSwapDestination: Identifiable, CaseIterable, Equatable, Hashable {
+  
+  static var allCases: [BuySendSwapDestination] = [.buy(), .send(), .swap()]
+  
+  case buy(BraveWallet.ERCToken? = nil)
+  case send(BraveWallet.ERCToken? = nil)
+  case swap(BraveWallet.ERCToken? = nil)
```

Comment:
If all of these take the same arg, it doesn't make sense to use an enum w/ associated value. Better to keep `BuySendSwapDestination` the same and put it inside a container `struct` i.e.

```swift
struct BuySendSwapDestination: Identifiable, Equatable, Hashable {
   enum Kind: Identifiable, CaseIterable, Equatable, Hashable { case buy, send, swap }
   var kind: Kind
   var initialToken: BraveWallet.ERCToken?
   var id: String { kind.id }
}
```